src/sentry/similarity/backends/metrics.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import absolute_import
+
+from sentry.similarity.backends.abstract import AbstractIndexBackend
+from sentry.utils.metrics import timer
+
+
+class MetricsWrapper(AbstractIndexBackend):
+    def __init__(self, backend, template='similarity.{}', scope_tag_name='scope'):
+        self.backend = backend
+        self.template = template
+        self.scope_tag_name = scope_tag_name
+
+    def __getattr__(self, name):
+        return getattr(self.backend, name)
+
+    def __instrumented_method_call(self, method, scope, *args, **kwargs):
+        with timer(self.template.format(method), tags={self.scope_tag_name: scope}):
+            return getattr(self.backend, method)(scope, *args, **kwargs)
+
+    def record(self, *args, **kwargs):
+        return self.__instrumented_method_call('record', *args, **kwargs)
+
+    def classify(self, *args, **kwargs):
+        return self.__instrumented_method_call('classify', *args, **kwargs)
+
+    def compare(self, *args, **kwargs):
+        return self.__instrumented_method_call('compare', *args, **kwargs)
+
+    def merge(self, *args, **kwargs):
+        return self.__instrumented_method_call('merge', *args, **kwargs)
+
+    def delete(self, *args, **kwargs):
+        return self.__instrumented_method_call('delete', *args, **kwargs)
+
+    def scan(self, *args, **kwargs):
+        return self.__instrumented_method_call('scan', *args, **kwargs)
+
+    def flush(self, *args, **kwargs):
+        return self.__instrumented_method_call('flush', *args, **kwargs)
+
+    def export(self, *args, **kwargs):
+        return self.__instrumented_method_call('export', *args, **kwargs)
+
+    def import_(self, *args, **kwargs):
+        return self.__instrumented_method_call('import_', *args, **kwargs)
```

Comment:
What's the justification for the `_`?

Reply:
Because `import` is a reserved word. So you can't name a function that.

Reply:
This guy Pythons. ☝️

Reply:
Could we not use `import` as the string supplied to `self.template.format(method...` though? I'm wondering about implications of leaving a trailing underscore on a metric path.

Reply:
No strong opinions, but personally I like the consistency with the implementation API and not having to special case the instrumentation wrapper for this one method, but I could be convinced otherwise. What implications are there?

Reply:
The blanket, "I don't know, computers not liking it? ¯\_(ツ)_/¯" so let's merge this and change if it the specific metric path doesn't get accepted because it doesn't end with an alphanumeric character.

Reply:
(To be clear, I think it looks like trash also and would rather not have the trailing underscore, but here we are.)